src/gsy_e/gsy_e_core/user_profile_handler.py

```diff
@@ -98,10 +102,13 @@ def get_first_week_from_profile(self, profile_uuid, current_timestamp) -> dict:
         """
         if not isinstance(profile_uuid, uuid.UUID):
             profile_uuid = uuid.UUID(profile_uuid)
-        first_datapoint_time = select(
+        first_datapoint = select(
             datapoint for datapoint in self.Profile_Database_ProfileTimeSeries
             if datapoint.profile_uuid == profile_uuid
-        ).order_by(lambda d: d.time).limit(1)[0].time
+        ).order_by(lambda d: d.time).limit(1)
+        if len(first_datapoint) == 0:
+            raise ProfileDBConnectionException("Profile in DB is empty.")
```

Comment:
I would also add the configuration uuid and / or at least the profile uuid. The latter would actually be the only thing necessary since we can deduce the configuration uuid from the profile uuid, however if it is possible to also add the config uuid for convenience would be great !

Reply:
I added the profile_uuid here: [e14b37c](https://github.com/gridsingularity/gsy-e/pull/1543/commits/e14b37c8e0195bdc8723f2657fd10a4765542a39)
The config_uuid is not that easy to inject into there. But you are right, it is easily acquired from the DB from the profile_uuid.